app-modules/notification/src/Notifications/Channels/MailChannel.php

```diff
@@ -140,15 +173,40 @@ public function send($notifiable, Notification $notification): void
                             : NotificationDeliveryStatus::Dispatched,
                         'quota_usage' => $quotaUsage,
                     ]);
+
+                    if ($emailMessage) {
+                        $emailMessage->quota_usage = $quotaUsage;
+
+                        $emailMessage->events()->create([
+                            'type' => (
+                                ($tenantMailConfig?->isDemoModeEnabled ?? false)
+                                && ((! $isSystemNotification) || (! $tenantMailConfig?->isExcludingSystemNotificationsFromDemoMode))
+                            )
+                                ? EmailMessageEventType::BlockedByDemoMode
+                                : EmailMessageEventType::Dispatched,
+                            'payload' => $result->toArray(),
+                            'occurred_at' => now(),
+                        ]);
```

Comment:
Can we clean up the condition here? Maybe that means duplicating the event creation code within `if` statements

Reply:
Let's sync on this one. You can let me know the direction you were thinking